pkg/bosh/manifest/cmd_instance_group_resolver.go

```diff
@@ -69,7 +69,10 @@ func (dg *InstanceGroupResolver) Manifest() (Manifest, error) {
 		return Manifest{}, err
 	}
 
-	return dg.manifest, nil
+	igManifest := dg.manifest
+	igManifest.InstanceGroups = []*InstanceGroup{dg.instanceGroup}
```

Comment:
Could we add some comments for this pr purpose?

Reply:
Not sure what you're looking for exactly. This pr essentially just makes the code behave like specified [in](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/rendering_templates.md#calculation-of-required-properties-for-an-instance-group-and-bpm-info) [the](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/controllers/extendedstatefulset.md#watches-in-sts-controller) [docs](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/controllers/extendedstatefulset.md#automatic-restart-of-containers)

Reply:
Not sure what you are looking for exactly. This PR essentially makes the code behave like specified [in](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/rendering_templates.md#calculation-of-required-properties-for-an-instance-group-and-bpm-info) [the](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/controllers/extendedstatefulset.md#watches-in-sts-controller) [docs](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/controllers/extendedstatefulset.md#automatic-restart-of-containers).

Reply:
Thx. 

So we have expected behavior that these versioned secrets bump especially only affected ig-resolved secrets. Is it right? I hope it does not conflict with our docs.

> - the bpm secrets are no longer bumped to identical versions
> - only affected ig-resolved secrets - and thus only affected instance groups - are bumped

Reply:
Yes, since `ig-resolved` secrets are supposed to only include the parts of the desired manifest that are relevant to one instance group they should only change if there are actual changes for it. And that's how it's specified imo, to quote the docs:
```
[The ig-resolved yaml] is the "Resolved Instance Group Properties" yaml file. It contains a deployment manifest structure that only has information pertinent to an instance group
```